### DISCUSSION TITLE
How to obtain the temperature of specified points in the steady-state simulation results of aedt-icepak

### DISCUSSION THREAD

**Discussion by YLSHA** [ansys/pyaedt#4355] [Q&A] [answered]
I'm using aedt-icepak for steady-state thermal simulations. 
Is there an API similar to Surface Probe in ansys-icepack? 
I checked the ipk.post, but it seems that I couldn't find the required feature.
I also tried to implement it with the monitor function (ipk.monitor.assign_point_monitor), but the number of monitors is too large (4000+), and the simulation freezes or gives an error.

**Comment by YLSHA** [comment]
I have found a solution with GUI. However, I didn't found the corresponding API in pyaedt, and the menu "automation-record script" did not record this step. The manual method is as follows
Create a point, select it along with the Temperature in the menu "Result-Fields Report-Data Table", and finally generate a report which gives me the point's temperature.

**Reply by YLSHA** [reply, marked as answer]
I found the solution and the problem seems to have been resolved, except for a very small number of points where the output temperature is 'NAN' (Could anyone tell me why? Is this related to mesh?). The solution for cube object is as follows 
```python
ipk.post.export_field_file_on_grid(
    quantity_name='Temperature',
    gridtype='Cartesian',
    grid_start=[0, 0, 0],                   #start point
    grid_stop=[1., 1., 0.5],               #stop point
    grid_step=[0.05, 0.05, 0.05],     #step for x/y/z coordinate
    )
```

**Reply by Samuelopez-ansys** [reply]
Hi @YLSHA ,

Right, this is the solution. If you have NaN values because there is no mesh on these points, you can just remove these points.

Regarding the recording, you are right, it is not recorded. Some Fields Calculator operations are not recorded, I am glad you found the solution.
